docs/proposals/operator-managed-infra.md

```diff
@@ -0,0 +1,128 @@
+* Make install onto existing K8s cluster straight forward and remove
+  complexity of using Terraform
+* Improved UX, all an end-user needs to do is install our manifests and ensure GCP service account has enough permissions
+* Enterprises generally have very specific requirements about how they create their K8s Clusters and nodepools.
+* Allow Substratus to work in a multi namespace environment. Currently namespace is hardcoded in workload identity settings in the terraform
+
+## How?
+
+<img src="../diagrams/identity-global.excalidraw.png"></img>
+
+### Installation
+**Installation Flow** will be responsible for the following:
+* Create a K8s cluster with nodes/GPUs, this step is optional if end-user already has a K8s cluster (eksctl or gcloud)
+* Create Bucket, this step is optional if end-user already has a bucket
+* Create Image Registry, this step is optional if end-user already has an image registry
+* REQUIRED: Create Google Service Account or AWS IAM Role that has resource level permissions to:
+  * read and write a specific bucket
+  * pull and push to a specific Image Registry (GAR/ECR)
+  * (GCP) set IAM policy on the service Account itself to be able to manage workload identity bindings to KSAs. This can be done by assigning the service account to be a ServiceAccountAdmin to itself
+  * (AWS) UpdateAssumeRolePolicy on the AWS Role used by Substratus (required for multi namespace support)
+* REQUIRED: Apply Substratus manifests and configure them to use correct bucket and image registry
+
+### Controller
+Controller will be responsible for the following:
+* Creating the K8s ServiceAccount (already done today in `enforceServiceAccount`), in addition should now also call `sci-${CLOUD}:BindKSAToIAMPrincipal` to ensure the new KSA has access to the AWS IAM Role/GCP Service Account
+
+* (GCP only) example set annotation for Google Service Account (already done today) AND call `sci-gcp:BindKSAToIAMPrincipal` to update IAM policy on the Service Account so it can use the Google Service Account. For example:
+  ```
+  gcloud iam service-accounts add-iam-policy-binding substratus@my-project.iam.gserviceaccount.com \
+   --role roles/iam.workloadIdentityUser \
+   --member "serviceAccount:myproject.svc.id.goog[new-namespace/substratus]"
+  ```
+* (AWS Only) Annotate the Kubernetes service account with ARN of IAM role AND (sci-aws) use `sci-aws:BindKSAToIAMPrincipal` to call `UpdateAssumeRolePolicy` to update the role's trust policy.
+* Any API calls made to clouds should go through a cloud manager e.g. `gcpmanager` or `awsmanager`
+
+### Substratus Cloud Interface
+The SCI will have a generic function `BindKSAToIAMPrincipal` that takes the following input:
+```
+Namespace: The K8s namespace
+KubernetesServiceAccount: The name of K8s Service Account
+IAMPrincipal: The GCP service account or IAM role
+Params Map[string]string: Any additional parameters that might be required by a specific cloud provider
```

Comment:
Prefer to omit Params until we find the need

Reply:
Defer to convo here: https://github.com/substratusai/substratus/pull/177

Reply:
OK with leaving this level of detail out of proposal, or just stating that these things are "first thoughts" in order to get proposal closed.

Reply:
I think when possible we should discuss it in the proposal instead of having to significantly change the code PR later. I will add a note saying the actual implementation will likely differ

Reply:
agreed with Nick that some of these should be provided as part of configMap or automatically detected at runtime instead of using params. This is such that the controller doesn't have to have different call for each cloud provider

Reply:
My take: proposals are good but probably shouldn't hammer out fine grain details. Proposals should paint the high-level picture with a rough outline of the implementation plan. Things like naming should often be deferred

Reply:
I think the risk with that is renaming in a PR with code sometimes requires full retesting and changing a lot of code. So I rather hash out as many details as possible. The interfaces and names of those interfaces should be discussed in the proposal when possible but not be required.